pkg/validating-webhooks/hiveextension/v1beta1/agentclusterinstall_admission_hook.go

```diff
@@ -133,6 +137,53 @@ func (a *AgentClusterInstallValidatingAdmissionHook) shouldValidate(admissionSpe
 	return true
 }
 
+func (a *AgentClusterInstallValidatingAdmissionHook) validateCreate(admissionSpec *admissionv1.AdmissionRequest) *admissionv1.AdmissionResponse {
+	contextLogger := log.WithFields(log.Fields{
+		"operation": admissionSpec.Operation,
+		"group":     admissionSpec.Resource.Group,
+		"version":   admissionSpec.Resource.Version,
+		"resource":  admissionSpec.Resource.Resource,
+		"method":    "validateCreate",
+	})
+
+	newObject := &hiveext.AgentClusterInstall{}
+	if err := a.decoder.DecodeRaw(admissionSpec.Object, newObject); err != nil {
+		contextLogger.Errorf("Failed unmarshaling Object: %v", err.Error())
+		return &admissionv1.AdmissionResponse{
+			Allowed: false,
+			Result: &metav1.Status{
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	// Add the new data to the contextLogger
+	contextLogger.Data["object.Name"] = newObject.Name
+
+	// verify that UserNetworkManagement is not set to false with SNO.
+	// if the user leave this field empty it is fine because the AI knows
+	// what to set as default
+	if isUserManagedNetworkingSetToFalseWithSNO(newObject) {
```

Comment:
This is a validation, wouldn't it be better to set a value instead of validating it? 
IMO it will be a better UX. 
For update it does make sense to validate, but in this case user create a CR without setting the obvious value

Reply:
It is forbidden to change the values inside a validation hook. There are mutating hooks for that and they are behaving in a special way that insure the proper outcome. As for the suggestion, the requirement specifically asked to validate the value and return error in case it is illegal as an example of the desired UX